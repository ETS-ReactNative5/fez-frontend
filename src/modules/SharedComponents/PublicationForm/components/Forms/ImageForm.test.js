jest.dontMock('./ImageFOrm');

import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import ImageForm from './ImageForm';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import PropTypes from 'prop-types';
import injectTapEventPlugin from 'react-tap-event-plugin';

function setup({submitting, isShallow = true}){

    const props = {
        submitting: submitting || false, // : PropTypes.bool,
    };

    if(isShallow) {
        return shallow(<ImageForm {...props} />);
    }

    return mount(<ImageForm {...props} />, {
        context: {
            muiTheme: getMuiTheme()
        },
        childContextTypes: {
            muiTheme: PropTypes.object.isRequired
        }
    });

}

beforeAll(() => {
    injectTapEventPlugin();
});

describe('ImageForm renders ', () => {
    it('component', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('component with 6 input fields', () => {
        const wrapper = setup({});
        expect(wrapper.find('Field').length).toEqual(6);
    });

    it('component with 5 required input fields', () => {
        const wrapper = setup({});
        expect(wrapper.find('Field .requiredField').length).toEqual(2);
        expect(wrapper.find('Field .requiredHintField').length).toEqual(1);
    });

    it('component with all fields disabled', () => {
        const wrapper = setup({submitting: true});
        wrapper.find('Field').forEach(field => {
            expect(field.props().disabled).toEqual(true);
        })
    });
});
