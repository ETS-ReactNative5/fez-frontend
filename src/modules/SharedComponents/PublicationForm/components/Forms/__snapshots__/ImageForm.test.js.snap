// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageForm renders  component 1`] = `
<div>
  <StandardCard
    className=""
    help={
      Object {
        "buttonLabel": "OK",
        "text": "Some text.",
        "title": "Image information",
      }
    }
    title="Image information"
  >
    <div
      className="columns"
    >
      <div
        className="column"
      >
        <Field
          autoFocus={true}
          className="requiredField"
          component={[Function]}
          disabled={false}
          floatingLabelText="Title"
          fullWidth={true}
          hintText="Title of image"
          name="rek_title"
          type="text"
          validate={
            Array [
              [Function],
            ]
          }
        />
      </div>
    </div>
    <div
      className="columns"
    >
      <div
        className="column"
      >
        <Field
          allowPartial={true}
          className="requiredHintField"
          component={[Function]}
          disabled={false}
          floatingTitle="Date created"
          floatingTitleRequired={true}
          name="rek_date"
          validate={
            Array [
              [Function],
            ]
          }
        />
      </div>
    </div>
    <div
      className="columns"
    >
      <div
        className="column"
      >
        <Field
          component={[Function]}
          disabled={false}
          floatingLabelText="Abstract"
          fullWidth={true}
          hintText="Please provide an abstract or summary of the work."
          multiLine={true}
          name="rek_description"
          rows={3}
          type="text"
        />
      </div>
    </div>
  </StandardCard>
  <StandardCard
    className=""
    help={
      Object {
        "buttonLabel": "OK",
        "text": "Enter authors in the order and form they appear on the published paper. Additional boxes will appear for more authors.",
        "title": "Creators",
      }
    }
    title="Creator name"
  >
    <div>
      Enter authors in the order and form they appear on the published paper. Additional boxes will appear for more authors.
    </div>
    <Field
      className="requiredField"
      component={[Function]}
      disabled={false}
      locale={
        Object {
          "form": Object {
            "locale": Object {
              "addButton": "Add creator",
              "identifierLabel": "UQ identifier (if available)",
              "nameAsPublishedHint": "Please type the name exactly as published",
              "nameAsPublishedLabel": "Creator's name as published",
            },
          },
          "header": Object {
            "locale": Object {
              "contributorAssignmentColumn": "Select your name",
              "deleteAll": "Remove all items",
              "deleteAllConfirmation": Object {
                "cancelButtonLabel": "No",
                "confirmButtonLabel": "Yes",
                "confirmationMessage": "Are you sure you want to delete all items?",
                "confirmationTitle": "Delete all",
              },
              "identifierColumn": "UQ identifier",
              "nameColumn": "creator's name as published",
              "reorderColumn": "Reorder items",
            },
          },
          "row": Object {
            "locale": Object {
              "deleteHint": "Remove this item",
              "deleteRecordConfirmation": Object {
                "cancelButtonLabel": "No",
                "confirmButtonLabel": "Yes",
                "confirmationMessage": "Are you sure you want to delete this item?",
                "confirmationTitle": "Delete item",
              },
              "moveDownHint": "Move item down the order",
              "moveUpHint": "Move item up the order",
              "ordinalData": Array [
                "First",
                "Second",
                "Third",
                "Fourth",
                "Fifth",
                "Sixth",
                "Seventh",
                "Eighth",
                "Ninth",
                "Tenth",
              ],
              "suffix": " listed creator",
            },
          },
        }
      }
      name="authors"
      showContributorAssignment={true}
      validate={
        Array [
          [Function],
        ]
      }
    />
  </StandardCard>
  <StandardCard
    className=""
    help={
      Object {
        "buttonLabel": "OK",
        "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam tempor, purus eget posuere fermentum, turpis magna auctor metus, in tincidunt est augue et orci.",
        "title": "Optional information",
      }
    }
    title="Optional information"
  >
    <div
      className="columns"
    >
      <div
        className="column"
      >
        <Field
          component={[Function]}
          disabled={false}
          floatingLabelText="Notes (not publicly viewable)"
          fullWidth={true}
          hintText="Please add any additional information here."
          multiLine={true}
          name="fez_record_search_key_notes.rek_notes"
          rows={1}
          type="text"
        />
      </div>
    </div>
    <div
      className="columns"
    >
      <div
        className="column"
      >
        <Field
          component={[Function]}
          disabled={false}
          floatingLabelText="Link"
          fullWidth={true}
          hintText="Enter URL for this work."
          name="rek_link"
          type="text"
          validate={
            Array [
              [Function],
            ]
          }
        />
      </div>
    </div>
  </StandardCard>
</div>
`;
